### PR TITLE
added parson submodule

### PR DESCRIPTION
**Pull request by PranavDhulipala**
unofficial parson library is not a part of the ros installation on ubuntu, edited the cmakelists to make use of the static library built from source.

### CODE REVIEW

File: roscpp_azure_iothub/CMakeLists.txt [hunk 1]
```diff
@@ -50,7 +45,11 @@ find_package(azure_iot_sdks CONFIG)
 if(azure_iot_sdks_FOUND)
   message(STATUS "azure-iot-sdk-c is found - ${IOTHUB_CLIENT_INCLUDES}")
 elseif(EXISTS "/usr/include/azureiot")
-  set(IOTHUB_CLIENT_INCLUDES "/usr/include/azureiot" "/usr/include/azureiot/inc")
+  if(MSVC)
```
Comment (ooeygui) [inline roscpp_azure_iothub/CMakeLists.txt:50]: MSVC should be false if there is a /usr directory. (/usr/include only exists on Linux)

File: roscpp_azure_iothub/CMakeLists.txt [hunk 2]
```diff
@@ -20,11 +20,6 @@ find_package(catkin REQUIRED COMPONENTS
   dynamic_reconfigure
 )
 
-find_package(unofficial-parson)
-if(unofficial-parson_FOUND)
-  message(STATUS "Found Parson")
-endif(unofficial-parson_FOUND)
-
 if(EXISTS "C:/opt/rosdeps/x64/include/azureiot")
```
Comment (ooeygui) [inline roscpp_azure_iothub/CMakeLists.txt]: Ack. This is the old location for rosdeps. This should come from vcpkg.

File: roscpp_azure_iothub/CMakeLists.txt [hunk 3]
```diff
@@ -50,7 +45,11 @@ find_package(azure_iot_sdks CONFIG)
 if(azure_iot_sdks_FOUND)
   message(STATUS "azure-iot-sdk-c is found - ${IOTHUB_CLIENT_INCLUDES}")
 elseif(EXISTS "/usr/include/azureiot")
-  set(IOTHUB_CLIENT_INCLUDES "/usr/include/azureiot" "/usr/include/azureiot/inc")
+  if(MSVC)
+    set(IOTHUB_CLIENT_INCLUDES "/usr/include/azureiot" "/usr/include/azureiot/inc")
+  else()
+    set(IOTHUB_CLIENT_INCLUDES "/usr/include/azureiot" "/usr/include/azureiot/inc" "parson")
```
Comment (ooeygui) [inline roscpp_azure_iothub/CMakeLists.txt:53]: Parson is a submodule, so it should be brought in when you recursively clone the repo correct?
I think this code implies that you are using the system parson, which would have to be installed prior to building.

**Review by ooeygui — CHANGES_REQUESTED** [review]

**Comment by ooeygui** [comment]
It also looks like your fork needs to be rebased.

**Comment by PranavDhulipala** [comment]
@ooeygui can you revert this pr? Turns out unofficial parson is not a part of ubuntu rather one has to install it manually. I changed my forked repo to include the original parson subfolder

**Pull request closed without merging**